File: Data Structures & Algorithms/buy-and-sell-crypto/submission-12.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if (prices.size() == 1) {
            return 0;
        }
        int max = 0;
        int left = 0;
        int right = 1;
        while (right < prices.size()) {
            while (prices[right] <= prices[left] && right < prices.size()) {
                left = right;
                right ++;
            }
            if (right < prices.size()) {
                max = std::max(max, prices[right] - prices[left]);
                right++;
            }
            
        }

        return max;
    }
};

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-12.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        if (len(nums) == 1):
            return nums[0];
        low = 0;
        high = len(nums) - 1
        
        while (high > low):
            mid = (high + low) // 2
            if (nums[mid] > nums[high]):
                # on the right side
                low = mid + 1
                # on the left side, including mid
                # nums[mid] if less than leftmost value
                # or equal (only happens if low = mid)
            else:
                high = mid;
        return nums[low]
        

            
        